blocks/term-tree-select/index.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External Dependencies
+ */
+import { unescape as unescapeString, repeat, reduce } from 'lodash';
+
+import SelectControl from '../inspector-controls/select-control';
+
+function getSelectOptions( terms, level = 0 ) {
+	return reduce( terms, ( options, term ) => (
```

Comment:
You might consider [Lodash's `flatMap`](https://lodash.com/docs/4.17.4#flatMap) which I recently found in #3185 to perform much better than `Array#reduce` + `Array#concat`.

Further, we can use array spread syntax to simplify this a bit:

```js
function getSelectOptions( terms, level = 0 ) {
	return flatMap( terms, ( term ) => [
		{
			value: term.id,
			label: repeat( '\u00A0', level * 3 ) + unescapeString( term.name ),
		},
		...getSelectOptions( term.children, level + 1 ),
	] );
}
```